### PR TITLE
[SYCL] Add more tests for experimental::printf

### PR DESCRIPTION
**Pull request by AlexeySachkov**

### CODE REVIEW

File: SYCL/Printf/helper.hpp [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#if defined(__SYCL_DEVICE_ONLY__) && defined(TEST_CONSTANT_AS)
+// On device side, we have to put format string into a constant address space
+// FIXME: remove this header completely once the toolchain's support for
+// non-constant address-spaced format string is stable.
```
Comment (AGindinson) [inline SYCL/Printf/helper.hpp]:
```suggestion
// generic address-spaced format strings is stable.
```

**Comment by AGindinson** [comment]
The Linux/Windows CI failures in `DeviceLib` and `Assert` tests are known & unrelated.

**Comment by AlexeySachkov** [comment]
I suggest we update CODEOWNERS for those new files as well

**Comment by AGindinson** [comment]
Together with the product PR, the unrelated DeviceLib [failure on Windows](http://llvm-ci-test2.intel.com:8080/blue/organizations/jenkins/SYCL_CI%2Fintel%2FWin%2FTest_Suite/detail/Test_Suite/496/pipeline) is the only one.

**Review by AlexeySachkov — COMMENTED** [review]
I can't approve because I'm the initial author of the PR, but changes introduced later by @AGindinson looks good to me.

**Comment by AGindinson** [comment]
@AlexeySachkov, could you please revise the latest changes?
@maksimsab, could you please take a look on behalf of the "non-authoring" DPC++ tools members?

**Review by maksimsab — APPROVED** [review]
LGTM

**Comment by AGindinson** [comment]
@intel/llvm-gatekeepers Could you please consider merging this based on the DPC++ tools member approval?

**Comment by vladimirlaz** [comment]
@AGindinson, @AlexeySachkov it looks like the new tests are failing on HIP in pre-commit: https://github.com/intel/llvm/runs/4643766953?check_suite_focus=true

Please fix it ASAP

**Comment by AGindinson** [comment]
@vladimirlaz, https://github.com/intel/llvm-test-suite/pull/685 should do the trick. Thanks!

**Pull request merged**